Sources/TensorFlow/Core/LazyTensorTFFunctionBuilder.swift

```diff
@@ -0,0 +1,315 @@
+    private let status: CTFStatus = TF_NewStatus()
+
+    /// Counter that is used for number the generated graph nodes.
+    private var nodeCounter: Int = 0
+
+    init(_ lazyTrace: LazyTensorTrace) {
+        var nodesCache: [ObjectIdentifier: CTFOperation?] = [:]
+        for op in lazyTrace.operations {
+            let opInputs = op.inputs.map { input -> TFGraph.Input in
+                switch input {
+                case LazyTensorOperation.Input.single(let h):
+                    return TFGraph.Input.single(makeTFOutput(h, nodesCache))
+                case LazyTensorOperation.Input.list(let elements):
+                    let tfInputs = elements.map { makeTFOutput($0, nodesCache) }
+                    return TFGraph.Input.list(tfInputs)
+                }
+            }
+            let graphNode = makeTFGraphNode(
+                name: op.name,
+                attrs: op.attributes,
+                inputs: opInputs,
+                device: op.deviceName)
+            nodesCache[ObjectIdentifier(op)] = graphNode
+            if op.name != "Placeholder" { nodes.append(graphNode) }
+        }
+        self.inputs = lazyTrace.inputs.map {
+            TF_Output(oper: nodesCache[ObjectIdentifier($0)]!, index: 0)
+        }
+        for output in lazyTrace.outputs {
+            let graphNode = nodesCache[ObjectIdentifier(output)]!
+            outputGroups.append(output.outputCount)
+            outputs += Array((0..<output.outputCount).map {
+                    TF_Output(oper: graphNode, index: Int32($0)) })
+        }
+    }
+
+    deinit {
+        TF_DeleteGraph(cTFGraph)
+        TF_DeleteStatus(status)
+    }
+
+    private func newNodeName(base: String) -> String {
+        let name = "\(base)_\(nodeCounter)"
+        nodeCounter += 1
+        return name
+    }
+
+    private func updateAttribute(
+        _ desc: CTFOperationDescription,
+        _ name: String,
+        _ attrValue: LazyTensorOperation.Attribute
+    ) {
+        switch attrValue {
+        case LazyTensorOperation.Attribute.tensorDataTypeValue(let value):
+            TF_SetAttrType(desc, name, value._cDataType)
+        case LazyTensorOperation.Attribute.boolValue(let value):
+            TF_SetAttrBool(desc, name, value ? 1 : 0)
+        case LazyTensorOperation.Attribute.intValue(let value):
+            TF_SetAttrInt(desc, name, Int64(value))
+        case LazyTensorOperation.Attribute.floatValue(let value):
+            TF_SetAttrFloat(desc, name, value)
+        case LazyTensorOperation.Attribute.doubleValue(let value):
+            TF_SetAttrFloat(desc, name, Float(value))
+        case LazyTensorOperation.Attribute.stringValue(let value):
+            value.utf8CString.withUnsafeBufferPointer { buffer in
+                // utf8CString is null-terminated; TF_SetAttrString wants
+                // non-null-terminated.
+                TF_SetAttrString(
+                    desc, name, buffer.baseAddress, buffer.count - 1)
```

Comment:
one argument per line when you break it. though I think this will all fit on one line.

Reply:
Done.